php/tests/generated_class_test.php

```diff
@@ -1342,4 +1342,37 @@ public function testArrayConstructor()
 
         TestUtil::assertTestMessage($m);
     }
+
+    #########################################################
+    # Test message equals.
+    #########################################################
+
+    public function testMessageEquals()
+    {
+        $m = new TestMessage();
+        TestUtil::setTestMessage($m);
+        $n = new TestMessage();
+        TestUtil::setTestMessage($n);
+        $this->assertEquals($m, $n);
+
+        $m = new TestMessage();
+        $m->setOptionalInt32(1);
+        $n = new TestMessage();
+        $n->setOptionalInt32(2);
+        $this->assertNotEquals($m, $n);
+
+        $m = new TestMessage();
+        $m->setOptionalString("a");
+        $n = new TestMessage();
+        $n->setOptionalString("b");
+        $this->assertNotEquals($m, $n);
+
+        $m = new TestMessage();
+        $subm = new Sub(['a' => 1]);
+        $m->setOptionalMessage($subm);
+        $n = new TestMessage();
+        $subn = new Sub(['a' => 2]);
+        $n->setOptionalMessage($subn);
+        $this->assertNotEquals($m, $n);
```

Comment:
I think it would be better to have two tests, `testMessageEquals` and `testMessageNotEquals`, which would have all the same tests here (for int, string, and nested message), but test both cases.